### ISSUE TITLE
Dequeued items are reversed when queue is Disposed without Flush

### ISSUE THREAD

**Issue by fosJoddie** [i-e-b/DiskQueue#5]
When queue Dispose is run, the list of dequeue operations to reinstate is fetched and iterated "in order".
This causes the items to be put back into the queue in wrong order
```
//original queue
//[ 1, 2, 3 ]

q.Dequeue()
q.Dequeue()

//queue now
//[3]

q.Dispose()

//queue after dispose
// [2, 1, 3]
```

fix:
in PersistentQueueImpl.cs line 369, change 
`from entry in reinstatedOperations` 
to 
`from entry in reinstatedOperations.Reverse()` 

This will cause the items to be reinstated in the order they were dequeued.

**Comment by i-e-b** [comment]
Thanks for that, I'll have a look as soon as I can.

**Comment by i-e-b** [comment]
Correction should be available on NuGet with version 1.2.0
https://www.nuget.org/packages/DiskQueue/1.2.0
